### PR TITLE
Update TravisCI config

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: tests/travis/install-extension-manager.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	cd $MW_INSTALL_PATH
+
+	if [ "$PHPUNIT" != "" ]
+	then
+		composer require 'phpunit/phpunit='$PHPUNIT --prefer-source --update-with-dependencies
+	else
+		composer require 'phpunit/phpunit=3.7.*' --prefer-source --update-with-dependencies
+	fi
+
+	if [ "$VERSION" != "" ]
```
Comment (mwjames) [inline tests/travis/install-extension-manager.sh]: Not sure if I run into trouble on Travis with `$VERSION` because it was already occupied.

**Comment by mwjames** [comment]
Using `DB` instead of `DBTYPE`

```
  - env: DBTYPE=mysql; MW=master
      php: 5.3
    - env: DBTYPE=mysql; MW=1.20.0
      php: 5.4
    - env: DBTYPE=sqlite; MW=1.22.0
      php: 5.6
    - env: DBTYPE=sqlite; MW=1.24.0
      php: hhvm
```

File: tests/travis/install-mediawiki.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+fi
+
+if [ "$DB" == "postgres" ]
+then
+  # See #458
+  sudo /etc/init.d/postgresql stop
+  sudo /etc/init.d/postgresql start
+
+  psql -c 'create database its_a_mw;' -U postgres
+  php maintenance/install.php --dbtype $DB --dbuser postgres --dbname its_a_mw --pass nyan TravisWiki admin --scriptpath /TravisWiki
```
Comment (mwjames) [inline tests/travis/install-mediawiki.sh:38]: $DB

**Comment by JeroenDeDauw** [comment]
Ah, I was just about to ask you for help :)

File: tests/travis/run-tests.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+set -ex
+
+BASE_PATH=$(pwd)
+MW_INSTALL_PATH=$BASE_PATH/../mw
+
+cd $MW_INSTALL_PATH/extensions/ExtensionManager
+
+if [ "$TYPE" == "coverage" ]
+then
+	composer phpunit -- --coverage-clover $BASE_PATH/build/coverage.clover
```
Comment (mwjames) [inline tests/travis/run-tests.sh]: Composer doesn't know `composer phpunit`

**Comment by mwjames** [comment]
Obvious failure:

```
Problem 1
    - Installation request for mediawiki/extension-manager dev-master -> satisfiable by mediawiki/extension-manager[dev-master].
    - mediawiki/extension-manager dev-master requires mwjames/service-registry ~0.1 -> no matching package found.
```

**Comment by JeroenDeDauw** [comment]
Thoughts on how to fix the obvious failure? I'm tempted to just include the files into this project for now and then refactor the code to stop using them

**Comment by mwjames** [comment]
> Thoughts on how to fix the obvious failure? I'm tempted to just include the files into this project for now and then refactor the code to stop using them

Yes, probably the best approach or I can create a quick composer package for the time being (and to be deleted later).

**Comment by JeroenDeDauw** [comment]
Muh. This setup is quite annoying when you change package dependencies. I removed it, though of course since this is not merged yet, it still fails.

**Comment by mwjames** [comment]
Travis can't find the extensions in `+cd ExtensionManager
./tests/travis/install-extension-manager.sh: line 31: cd: ExtensionManager: No such file or directory
` because it is installed in `/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager`

```
  - Installing mediawiki/extension-manager (dev-master afd0a51)
    Cloning afd0a5190f6a199ffd753ad07c8c89daa560d3b8
Executing command (CWD): git clone --no-checkout 'git://github.com/SemanticMediaWiki/ExtensionManager.git' '/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager' && cd '/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager' && git remote add composer 'git://github.com/SemanticMediaWiki/ExtensionManager.git' && git fetch composer
Executing command (/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager): git remote set-url --push origin 'git@github.com:SemanticMediaWiki/ExtensionManager.git'
Executing command (/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager): git branch -r
Executing command (/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager): git checkout 'master'
Executing command (/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager): git reset --hard 'afd0a5190f6a199ffd753ad07c8c89daa560d3b8'
    REASON: Required by root: Install command rule (install mediawiki/extension-manager dev-master)
Executing command (/home/travis/build/SemanticMediaWiki/mw/vendor/mediawiki/extension-manager): git log -n1 --pretty=%ct 'afd0a5190f6a199ffd753ad07c8c89daa560d3b8'
Writing lock file
Generating autoload files
+ls
api.php        Gemfile.lock   load.php5         resources
api.php5       Gruntfile.js   LocalSettings.php     serialized
autoload.php   HISTORY        maintenance       skins
cache          images         mw-config         StartProfiler.sample
composer.json  img_auth.php   opensearch_desc.php   tests
composer.lock  img_auth.php5  opensearch_desc.php5  thumb_handler.php
COPYING        includes       package.json      thumb_handler.php5
CREDITS        index.php      profileinfo.php       thumb.php
docs           index.php5     profileinfo.php5      thumb.php5
extensions     INSTALL        README            UPGRADE
FAQ        languages      README.mediawiki      vendor
Gemfile        load.php       RELEASE-NOTES-1.25    wiki.phtml
+cd extensions
+ls
README
+cd ExtensionManager
./tests/travis/install-extension-manager.sh: line 31: cd: ExtensionManager: No such file or directory
```

https://travis-ci.org/SemanticMediaWiki/ExtensionManager/jobs/51616207#L346

The cause was a missing `composer/installers` dependency.

**Pull request merged**